enhancements/network/multi-network-vrfs.md

```diff
@@ -0,0 +1,830 @@
+and then set the active network on the namespace via an annotation: ```k8s.ovn.org/active-network: l3-network```. If
+pods existed before the NAD was created, errors will be logged, and the active-network will remain “default”.
+
+After creating the NAD, pods created in this namespace will connect to the newly defined network as their primary
+network. The primary key is used so that OVN-Kubernetes knows which network should be used, in case there are multiple
+NADs created for a namespace (secondary networks).
+
+After a pod is created with the default-network annotation, it will then be annotated by OVN-Kubernetes with the
+appropriate networking config:
+
+```
+trozet@fedora:~/Downloads$ oc get pods -o yaml -n ns1
+apiVersion: v1
+items:
+- apiVersion: v1
+  kind: Pod
+  metadata:
+    annotations:
+      k8s.ovn.org/pod-networks: '{"default":{"ip_addresses":["10.244.0.4/24"],"mac_address":"0a:60:0a:f4:00:04","gateway_ips":["10.244.0.1"],"routes":[{"dest":"10.96.0.2/32","nextHop":"10.244.0.1"},{"dest":"10.96.0.3/32","nextHop":"10.244.0.1"}]},"ns1/l3-network":{"type":primary,"ip_addresses":["10.128.0.4/24"],"mac_address":"0a:58:0a:80:00:04","routes":[{"dest":"10.128.0.0/16","nextHop":"10.128.0.1"}],"ip_address":"10.128.0.4/24"}}'
+      k8s.v1.cni.cncf.io/network-status: |-
+        [{
+            "name": "default",
+            "interface": "eth0",
+            "ips": [
+                "10.244.0.4"
+            ]},
+            {
+            "name": "ns1/l3-network",
+            "interface": "eth1",
+            "ips": [
+                "10.128.0.4"
+            ]},
+            {
+            "mac": "0a:58:0a:f4:02:03",
+            "default": true,
+            "dns": {}
+        }]
+status:
+  phase: Running
+  podIP: 10.244.0.4
+  podIPs:
+  - ip: 10.244.0.4
+```
+
+Multiple namespaces may also be configured to use the same network. In this case the underlying OVN network will be the
+same, following a similar pattern to what is [already supported today for secondary networks](https://docs.openshift.com/container-platform/4.15/networking/multiple_networks/configuring-additional-network.html#configuration-ovnk-network-plugin-json-object_configuring-additional-network).
+
+### CRDs for Managing Networks
+
+Network Attachment Definitions (NADs) are the current way to configure the network in OVN-Kubernetes today, and the
+method proposed in this enhancement. There are two major shortcomings of NAD:
+
+1. It has free-form configuration that depends on the CNI. There is no API validation of what a user enters, leading to
+mistakes which are not caught at configuration time and may cause unexpected functional behavior at runtime.
+2. It requires cluster admin RBAC in order to create the NAD.
+
+In order to address these issues, a proper CRD may be implemented which indirectly creates the NAD for OVN-Kubernetes.
+This solution may consist of more than one CRD, namely an Admin based CRD and one that is namespace scoped for tenants.
```

Comment:
I think this is a practical approach. Since net-attach-defs are kind of inherently privileged... Having an abstraction that's allowed at the namespace level in order to kind of isolate these tenants at the namespace level without requiring them to be cluster admin.

I hate more abstraction, but, I guess the other part is... with this abstraction the net-attach-def is created given the user's expression in the CRD, which means you can validate it, too.

So, I'm into this approach.